phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixEmbeddedDriver.java

```diff
@@ -199,15 +201,20 @@ public void close() throws SQLException {
         private static final Object KERBEROS_LOGIN_LOCK = new Object();
         private static final char WINDOWS_SEPARATOR_CHAR = '\\';
         private static final String REALM_EQUIVALENCY_WARNING_MSG = "Provided principal does not contan a realm and the default realm cannot be determined. Ignoring realm equivalency check.";
+
         private static SQLException getMalFormedUrlException(String url) {
             return new SQLExceptionInfo.Builder(SQLExceptionCode.MALFORMED_CONNECTION_URL)
             .setMessage(url).build().buildException();
         }
-        
-		public String getZookeeperConnectionString() {
-			return getZookeeperQuorum() + ":" + getPort();
-		}
-        
+
+        private static SQLException getMalFormedUrlException(String message, String url) {
+            return getMalFormedUrlException(message + "\n" + url);
+        }
+
+        public String getZookeeperConnectionString() {
```

Comment:
This is used by Tephra only, which won't be able to make sense of a HRpc quorum.
Better add a check here, and fail early if the boostrap is not ZK.
(or maybe figure out the cluster ZK quorum somehow, it is possible)